src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -51,35 +31,30 @@ export interface CreateTransactionArgs {
   delayExecution?: boolean
 }
 
-type RequiredTxProps = CreateTransactionArgs &
-  Required<Pick<CreateTransactionArgs, 'txData' | 'operation' | 'navigateToTransactionsTab' | 'origin'>>
-
-type CreateTransactionAction = ThunkAction<Promise<void | string>, AppReduxState, DispatchReturn, AnyAction>
-type ConfirmEventHandler = (safeTxHash: string) => void
-type ErrorEventHandler = () => void
-
-export const METAMASK_REJECT_CONFIRM_TX_ERROR_CODE = 4001
-
-const navigateToTx = (safeAddress: string, txDetails: TransactionDetails) => {
-  if (!isMultiSigExecutionDetails(txDetails.detailedExecutionInfo)) {
-    return
-  }
-  const prefixedSafeAddress = getPrefixedSafeAddressSlug({ shortName: extractShortChainName(), safeAddress })
-  const txRoute = generatePath(SAFE_ROUTES.TRANSACTIONS_SINGULAR, {
-    [SAFE_ADDRESS_SLUG]: prefixedSafeAddress,
-    [TRANSACTION_ID_SLUG]: txDetails.detailedExecutionInfo.safeTxHash,
-  })
-
-  history.push(txRoute)
+export type CreateTransactionAction = ThunkAction<Promise<void | string>, AppReduxState, DispatchReturn, AnyAction>
+export type ConfirmEventHandler = (safeTxHash: string) => void
+export type ErrorEventHandler = () => void
+
+const getCreationTxProps = ({
+  txData = EMPTY_DATA,
+  operation = Operation.CALL,
+  navigateToTransactionsTab = true,
+  origin = null,
+  ...rest
+}: CreateTransactionArgs): RequiredTxProps => {
+  return { txData, operation, navigateToTransactionsTab, origin, ...rest }
 }
 
-const getSafeTxGas = async (txProps: RequiredTxProps, safeVersion: string): Promise<string> => {
+const getSafeTxGas = async (
```

Comment:
Is this function duplicated? I think there was a folder with gas-related logic

Reply:
No, but it's using the `estimateSafeTxGas` function from the gas-related logic folder.